packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-dependencies.js

```diff
@@ -155,3 +192,129 @@ async function writeNativeDependenciesJson(nativeDependencies, outputDir, widget
 async function asyncWhere(array, filter) {
     return (await Promise.all(array.map(async el => ((await filter(el)) ? [el] : [])))).flat();
 }
+
+async function scanDependency(dir) {
+    let pkg = null;
+    const pkgJsonPath = join(dir, "package.json");
+    const exists = existsSync(pkgJsonPath);
+    if (!exists) {
+        return null;
+    }
+    const pkgJson = JSON.parse(readFileSync(pkgJsonPath));
+    const license = pkgJson.license || pkgJson.licenses;
+    const hasLicense = license && license.length > 0;
+    const name = pkgJson.name;
+    if (!name && !hasLicense) {
+        return null;
+    }
+    pkg = pkgJson;
+    const absolutePath = join(dir, "licen[cs]e");
+    const licenseFile = (await fg([absolutePath], { cwd: dir, caseSensitiveMatch: false }))[0];
+    if (licenseFile) {
+        pkg.licenseText = readFileSync(licenseFile, "utf-8");
+    }
+    return pkg;
+}
+
+class LicensePlugin {
+    constructor(options = {}) {
+        this._options = options;
+        this._dependencies = {};
+    }
+
+    addDependency(pkg, { isTransitive }) {
+        const name = pkg.name;
+        if (!name) {
+            console.warn("Trying to add dependency without any name, skipping it!");
+        } else if (!_.has(this._dependencies, name)) {
+            this._dependencies[name] = new Dependency({ ...pkg, ...{ isTransitive } });
+        }
+    }
+
+    config() {
+        const thirdParty = this._options.thirdParty;
+        if (!thirdParty) {
+            return;
+        }
+
+        const outputDependencies = _.chain(this._dependencies).values().value();
+        const output = thirdParty.output;
+
+        if (output) {
+            this._exportThirdParties(outputDependencies, output);
+        }
+    }
+
+    _exportThirdParties(dependencies, outputs) {
```

Comment:
This "private" method is only referenced once, by another private method. I think it makes sense it inline it.

Reply:
done